rfcs/45-stageless.md

```diff
@@ -0,0 +1,584 @@
+
+If you just want to run your game logic systems in the middle of your schedule, after input is processed but before rendering occurs, add the premade `CoreLabel::AppLogic` to them.
+
+### Label configuration
+
+Each system label has its own associated `SystemConfig`, stored in the corresponding `Schedule`.
+This configuration is applied in an additive fashion to each system with that label.
+
+You can define labels in a standalone fashion, configuring them at the time of creation:
+
+```rust
+#[derive(SystemLabel)]
+enum Physics {
+    Forces,
+    CollisionDetection,
+    CollisionHandling,
+}
+
+impl Plugin for PhysicsPlugin{
+
+    fn build(app: &mut App){
+        // Within each frame, physics logic needs to occur after input handling, but before rendering
+        let mut common_physics_config = SystemConfig::new().after(CoreLabel::InputHandling).before(CoreLabel::Rendering);
+
+        app
+        // We can reuse this shared configuration on each of our labels
+        .configure_label(Physics::Forces.configure(common_physics_config).before(Physics::CollisionDetection))
+        .configure_label(Physics::CollisionDetection.configure(common_physics_config).before(Physics::CollisionHandling))
+        .configure_label(Physics::CollisionHandling.configure(common_physics_config))
+        // And then apply that config to each of the systems that have this label
+        .add_system(gravity.label(Physics::Forces))
+        // These systems have a linear chain of ordering dependencies between them
+        // Systems earlier in the chain must run before those later in the chain
+		// Other systems can run in between these systems;
+		// use `add_atomic_system_chain` if this is not desired
+        .add_system_chain([broad_pass, narrow_pass].label(Physics::CollisionDetection))
+        // System sets apply a set of labels to a collection of systems
+        // and are helpful for reducing boilerplate
+        .add_system_set([compute_forces, collision_damage], Physics::CollisionHandling);
+    }
+}
+```
+
+You can apply the same label(s) to many systems at once using **system sets** with the `App::add_system_set(systems: impl SystemIterator, labels: impl SystemLabelIterator)` method.
+
+### System ordering
+
+The most important way we can configure systems is by telling the scheduler *when* they should be run.
+The ordering of systems is always defined relative to other systems: either directly or by checking the systems that belong to a label.
+
+There are two basic forms of system ordering constraints:
+
+1. **Strict ordering constraints:** `strictly_before` and `strictly_after`
+   1. A system cannot be scheduled until "strictly before" systems must have been completed during this iteration of the schedule.
+   2. Simple and explicit.
+   3. Can cause unneccessary blocking, particularly when systems are configured at a high-level.
+2. **If-needed ordering constraints:** `.before` and `.after`

```

Comment:
Emphasize that this is indistinguishable from it being scheduled after.

Reply:
Unless I'm misunderstanding this is'nt entirely true. If a system `a` has commands and a system `b` is `before` `a`, then even if `a` doesn't conflict with `b` they won't necessarily commute, if there happens to be a `flush_commands` running between them.

Reply:
This has been made transitive.